.github/workflows/release.yml

```diff
@@ -46,12 +46,19 @@ jobs:
       id-token: write
     steps:
       - name: Download packages
-        uses: actions/download-artifact@v3
+        uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16  # v4.1.8
+        with:
+          pattern: packages-*
+          path: dist
+          merge-multiple: true
 
-      - name: Consolidate packages for upload
-        run: |
-          mkdir dist
-          cp packages-*/* dist/
+      - name: Print out packages
+        run: ls dist
+
+      - name: Generate artifact attestation for sdist and wheel
+        uses: actions/attest-build-provenance@ef244123eb79f2f7a7e75d99086184180e6d0018  # v1.4.4
+        with:
+          subject-path: dist/cycler-*
 
       - name: Publish package distributions to PyPI
-        uses: pypa/gh-action-pypi-publish@b7f401de30cb6434a1e19f805ff006643653240e # v1.8.10
+        uses: pypa/gh-action-pypi-publish@15c56dba361d8335944d31a2ecd17d700fc7bcbc  # v1.12.2
```

Comment:
Current version is 1.12.3: https://github.com/pypa/gh-action-pypi-publish/releases/tag/v1.12.3